src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/DefaultHtmlGenerator.cs

```diff
@@ -106,6 +107,7 @@ public string FormatValue(object value, string format)
 
         /// <inheritdoc />
         public virtual TagBuilder GenerateActionLink(
+            ViewContext viewContext,
```

Comment:
> Everything that calls this should already have access to a constructed `UrlHelper`

Everything? Perhaps most of the tag helper call sites. But where is this `IUrlHelper` instance in `HtmlHelper`?